### ISSUE TITLE
False positive: unneeded not

### ISSUE THREAD

**Issue by clo-vis** [pylint-dev/pylint#8890]
### Bug description

```python
a = {1: 1}
b = {2: 2}
if not a.keys() < b.keys():
    print("!")
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:3:3: C0113: Consider changing "not a.keys() < b.keys()" to "a.keys() >= b.keys()" (unneeded-not)
```

### Expected behavior

No error message, as for sets (not A < B) is not the same as (B >= A).

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.6
Python 3.10.4
```

**Comment by nickdrozd** [comment]
Flipping `not <` to `>=` is valid just when the relation is a [total order](https://en.wikipedia.org/wiki/Total_order) -- when for any two distinct values `a` and `b`, either `a < b` or `b > a`. This is true for the `<` relation over `int`, and also over `list` (where `<` denotes [lexicographic ordering](https://en.wikipedia.org/wiki/Lexicographic_order)).

But for sets, `<` denotes the [subset relation](https://en.wikipedia.org/wiki/Subset), which is a [partial order](https://en.wikipedia.org/wiki/Partial_order). So `{1} < {2}` and `{1} > {2}` are both false.

The check should only trigger when the relation can be confirmed to be a total order (`int`, `list`, etc).

**Comment by nickdrozd** [comment]
Looks like the check does work correctly for sets, but not for `dict_keys`, which are not sets but are set-like.
